test/kernels/test_qkernel.py

```diff
@@ -515,5 +516,122 @@ def test_user_parameters(self):
             self.assertEqual(list(qkclass.user_param_binds.keys()), qkclass.user_parameters)
 
 
+class TestQuantumKernelBatching(QiskitMachineLearningTestCase):
+    """Test QuantumKernel circuit batching
+
+    Creates a class factory that takes the QuantumInstance class,
+    applies a method decorator to the execute function and returns the modified class.
+    The decorator function records the actual number of circuits being passed
+    to execute does not exceed the batch_size requested by the Quantum Kernel.
+    Also checks that the sum of the batch sizes matches the total number of expected
+    circuits.
+    Checks batching with both statevector simulator and QASM simulator.
+    """
+
+    def count_circuits(self, func):
+        """Wrapper to record the number of circuits passed to QuantumInstance.execute.
+
+        Args:
+            func (Callable): execute function to be wrapped
+
+        Returns:
+            Callable: function wrapper
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwds):
+            self.circuit_counts.append(len(args[1]))
+            return func(*args, **kwds)
+
+        return wrapper
+
+    def setUp(self):
+        super().setUp()
+
+        algorithm_globals.random_seed = 10598
+
+        self.shots = 12000
+        self.batch_size = 3
+        self.circuit_counts = []
+
+        QuantumInstance.execute = self.count_circuits(QuantumInstance.execute)
```

Comment:
This is changing the loaded class itself not an instance. This will affect any use later, in any other tests, by anything using this class. This is not a good idea. Now you can monkey patch the instance itself, rather than the class, after its created below.

Reply:
Yep, excellent point. I've now monkey patched the two instances rather than the class.